### PR TITLE
Deployments: Set up new layout

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: pkg/deployments/hardhat.config.ts [hunk 1]
```diff
@@ -1,144 +1,18 @@
...
-    compilers: hardhatBaseConfig.compilers,
-    overrides: { ...hardhatBaseConfig.overrides(name) },
-  },
-  tenderly: {
-    username: 'balancer',
-    project: 'v2',
-  },
+task('deploy', 'Run deployment task')
+  .addParam('task', 'Deployment task ID')
+  .addOptionalParam('force', 'Whether the task to deploy must be ignore previous deployments or not')
```
Comment (nventuro) [inline pkg/deployments/hardhat.config.ts:12]:
```suggestion
  .addOptionalParam('force', 'Ignore previous deployments')
```

File: pkg/deployments/tasks/20210418-vault/input.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  weth: string;
+  pauseWindowDuration: number;
+  bufferPeriodDuration: number;
+};
+
+const authorizer = new Task('20210418-authorizer');
+
+export default {
+  authorizer,
+  weth: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
```
Comment (nventuro) [inline pkg/deployments/tasks/20210418-vault/input.ts:14]: I'm still understanding how this is structured, but note that this address is only valid for mainnet.

File: pkg/deployments/tasks/20210418-vault/input.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  pauseWindowDuration: 7776000,
+  bufferPeriodDuration: 2592000,
```
Comment (nventuro) [inline pkg/deployments/tasks/20210418-vault/input.ts:16]: 3 * MONTH, MONTH, etc

File: pkg/deployments/tasks/20210418-vault/output/goerli.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "vault": "0xBA12222222228d8Ba445958a75a0704d566BF2C8",
+  "balancerHelpers": "0x5aDDCCa35b7A0D07C74063c48700C8590E87864E"
```
Comment (nventuro) [inline pkg/deployments/tasks/20210418-vault/output/goerli.json:3]: Wouldn't it make sense to have a separate deployment for the helpers?

**Review by nventuro — COMMENTED** [review]
I don't yet grasp the full picture as `Task.ts` is missing in this branch, but this looks extremely promising :star_struck:

**Review by albertoblue87-netizen — COMMENTED** [review]
Hecho

**Pull request merged**